File: puzzles/ts/power-of-thor1/power-of-thor1.ts
const inputs: string[] = readline().split(' ');
const lightX: number  = Number(inputs[0]); // the X position of the light of power
const lightY: number = Number(inputs[1]); // the Y position of the light of power
let thorX: number = Number(inputs[2]); // Thor's starting X position
let thorY: number = Number(inputs[3]); // Thor's starting Y position

// game loop
while (true) {
  let move: string = "";
  if (thorY < lightY) {
    move += 'S';
    thorY += 1;
  } else if (thorY > lightY) {
    move += 'N';
    thorY -= 1;
  }
  if (thorX < lightX) {
    move += 'E';
    thorX += 1;
  } else if (thorX > lightX) {
    move += 'W';
    thorX -= 1;
  }
  // A single line providing the move to be made: N NE E SE S SW W or NW
  console.log(move);
}

File: puzzles/kotlin/src/power-of-thor1/power-of-thor1.kt
import java.util.*

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)
  val lightX = input.nextInt() // the X position of the light of power
  val lightY = input.nextInt() // the Y position of the light of power
  var x = input.nextInt() // Thor's starting X position
  var y = input.nextInt() // Thor's starting Y position

  // game loop
  while (true) {
    val remainingTurns = input.nextInt() // The remaining amount of turns Thor can move. Do not remove this line.
    var move = StringBuilder()

    if (y < lightY) {
      move.append("S")
      y++
    } else if (y > lightY) {
      move.append("N")
      y--
    }

    if (x < lightX) {
      move.append("E")
      x++
    } else if (x > lightX) {
      move.append("W")
      x--
    }

    // A single line providing the move to be made: N NE E SE S SW W or NW
    println(move)
  }
}
